packages/components/src/button/test/deprecated-props.js

```diff
@@ -0,0 +1,98 @@
+/**
```

Comment:
I think this file can just be a test case in main `test/index.js` file. Don't need to create another test file.

Reply:
The reason I created a separate file was to be able to mock the `@wordpress/deprecate` package without affecting the other `<Button />` tests. But thanks to your suggestion to use `expect( console ).toHaveWarned()`, that's not necessary anymore.

I went ahead and moved the tests back into `test/index.js` in https://github.com/WordPress/gutenberg/pull/31713/commits/69ed9efd0c6674c8e7d627dc1971e875081bd4fa